### ISSUE TITLE
Multiple Calls to OnConnectedAsync with the same connectionId

### ISSUE THREAD

**Issue by cheekyseagull** [SignalR/SignalR#436]
Here is the conversation I had with Damien Edwards regarding this issue:

abraham.fine
Hello. Is it possible that OnConnectedAsync is called multiple times for the same connectionId?
 DamianEdwards
@abraham.fine possible: yes, although we try to prevent it
11:52:51 AM
that is, the idea is it only gets called once
11:53:01 AM
but
11:53:11 AM
 DamianEdwards
if a transport connect times out
11:53:11 AM
then it could happen again on transport fallback
11:53:17 AM
actually we could probably handle that better
11:53:32 AM
'cos once the connect event is raised, it's in our tracking store
11:53:42 AM
so if a connect comes in again for a different transport it shouldn't raise the event
11:53:55 AM
i'd have to check the code
11:54:02 AM
either way, worth logging an issue
11:55:00 AM
 abraham.fine
ok, it happens while running crank
11:55:58 AM
so i can't imagine that the transport connect would time out, given it occurs in the same batch of 100 connections
11:56:46 AM
i have to run for an hour, i will be back to keep discussing this issue

Long story short, once a connection is established in Crank, multiple calls to OnConnectedAsync are made with the same connectionId (this occurs if I am generating my own connectionId or using the stock generator).
